packages/pn-commons/src/components/NotificationDetail/NotificationPaymentRecipient.tsx

```diff
@@ -1,55 +1,129 @@
+  handleDownloadAttachamentPagoPA,
+  handleReloadPayment,
+}) => {
+  const [selectedPayment, setSelectedPayment] = useState<PagoPAPaymentHistory | null>(null);
+  const isSinglePayment =
+    payments.length === 1 && payments[0].pagoPA?.status === PaymentStatus.REQUIRED;
+
+  const handleClick = (event: React.ChangeEvent<HTMLInputElement>) => {
+    const radioSelection = event.target.value;
+
+    setSelectedPayment(
+      payments.find((item) => item.pagoPA?.noticeCode === radioSelection)?.pagoPA || null
+    );
+  };
+
+  const handleDeselectPayment = () => {
+    setSelectedPayment(null);
+  };
+
+  useEffect(() => {
+    if (isSinglePayment) {
+      setSelectedPayment(payments[0].pagoPA || null);
+    }
+  }, [isSinglePayment, payments]);
+
   return (
-    <Fragment>
+    <Box display="flex" flexDirection="column" gap={2}>
       <Typography variant="h6" data-testid="notification-payment-recipient-title">
-        {getLocalizedOrDefaultLabel('notifications', 'detail.payment.title', 'Pagamento')}
+        {getLocalizedOrDefaultLabel('notifications', 'detail.payment.title')}
       </Typography>
-      <Typography
-        variant="body2"
-        sx={{ mt: 2 }}
-        data-testid="notification-payment-recipient-subtitle"
-      >
-        {getLocalizedOrDefaultLabel(
-          'notifications',
-          'detail.payment.subtitle',
-          'In questa notifica ci sono più avvisi di pagamento: seleziona quello che vuoi pagare. Alcuni avvisi includono i costi di notifica.'
-        )}
+      <Typography variant="body2" data-testid="notification-payment-recipient-subtitle">
+        {getLocalizedOrDefaultLabel('notifications', 'detail.payment.subtitle')}
         &nbsp;
         <Link href={void 0} target="_blank" fontWeight="bold" sx={{ cursor: 'pointer' }}>
-          {getLocalizedOrDefaultLabel('notifications', 'detail.payment.how', 'Come mai?')}
+          {getLocalizedOrDefaultLabel('notifications', 'detail.payment.how')}
         </Link>
       </Typography>
-
-      <Button fullWidth variant="contained" sx={{ mt: 2 }} data-testid="pay-button">
-        {getLocalizedOrDefaultLabel('notifications', 'detail.payment.pay', 'Paga')}
+      <Box>
+        <RadioGroup name="radio-buttons-group" value={selectedPayment} onChange={handleClick}>
+          {payments.map((payment, index) =>
+            payment.pagoPA ? (
+              <Box mb={2} key={`payment-${index}`} data-testid="pagopa-item">
+                <NotificationPaymentPagoPAItem
+                  pagoPAItem={payment.pagoPA}
+                  loading={loading}
+                  isSelected={payment.pagoPA.noticeCode === selectedPayment?.noticeCode}
+                  handleReloadPayment={() => handleReloadPayment([payment])}
+                  handleDeselectPayment={handleDeselectPayment}
+                  isSinglePayment={isSinglePayment}
+                />
+              </Box>
+            ) : null
+          )}
+        </RadioGroup>
+      </Box>
+      <Button
```

Comment:
if we have all payments PAID, how does it work? maybe we have to hide the button

Reply:
According to Figma we have to hide all buttons if all payments is paid.